src/nrtk_explorer/library/dataset.py

```diff
@@ -1,24 +1,54 @@
-import kwcoco
-from pathlib import Path
-
-
-def load_dataset(path: str):
-    return kwcoco.CocoDataset(path)
-
+"""
+Module to load the dataset and get the image file path given an image id.
 
-dataset: kwcoco.CocoDataset = kwcoco.CocoDataset()
-dataset_path: str = ""
-
-
-def get_dataset(path: str, force_reload=False):
-    global dataset, dataset_path
-    if dataset_path != path or force_reload:
-        dataset_path = path
-        dataset = load_dataset(dataset_path)
-    return dataset
+Example:
+    dataset = get_dataset("path/to/dataset.json")
+    image_fpath = dataset.get_image_fpath("image_id")
+"""
 
+from functools import lru_cache
+from pathlib import Path
 
-def get_image_path(id: str):
-    dataset_dir = Path(dataset_path).parent
-    file_name = dataset.imgs[int(id)]["file_name"]
```

Comment:
The `int` cast is key.  I'm getting error when I click on a point:
```
Traceback (most recent call last):
  File "/home/paul/src/cdao/.venv/lib/python3.10/site-packages/trame_server/utils/asynchronous.py", line 19, in handle_task_result
    task.result()
  File "/home/paul/src/cdao/nrtk-explorer/src/nrtk_explorer/app/transforms.py", line 298, in _update_images
    filename = get_dataset(self.state.current_dataset).get_image_fpath(selected_id)
  File "/home/paul/src/cdao/.venv/lib/python3.10/site-packages/kwcoco/coco_dataset.py", line 344, in get_image_fpath
    gpath = ub.Path(self.bundle_dpath) / img['file_name']
TypeError: string indices must be integers
```

Reply:
Thanks for pointing it out, I am trying to select single points in the embeddings plot but I am not hitting this error. Is there anything else that you did.

Reply:
I have kwcoco installed =)

Reply:
aha! found and resolved by making selected_id int, and casting to int during invokation